### PR TITLE
[kotlin] Add json annotation to each enum value.

### PR DESCRIPTION
**Pull request by ejithon**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [x] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

https://github.com/swagger-api/swagger-codegen/issues/7811

Add moshi.Json annotation to each enum value.

### CODE REVIEW

**Comment by wing328** [comment]
cc @jimschubert for review.

File: modules/swagger-codegen/src/main/resources/kotlin-client/data_class.mustache [hunk 1]
```diff
@@ -1,3 +1,5 @@
+{{#hasEnums}}import com.squareup.moshi.Json{{/hasEnums}}
+
```
Comment (jimschubert) [inline modules/swagger-codegen/src/main/resources/kotlin-client/data_class.mustache]: Can you remove this extra newline?

Also, I think the newline generated by the hasEnums check when that is false can be eliminated by putting the opening and closing tags on their own lines.

Comment (ejithon) [reply modules/swagger-codegen/src/main/resources/kotlin-client/data_class.mustache]: @jimschubert Thank you for reviewing, I will remove it soon.

**Comment by ejithon** [comment]
@jimschubert Does my [updating](https://github.com/swagger-api/swagger-codegen/pull/7908/commits/9e4d7d286c6c5fb4640f090a0cf3925756c7bfa1) make sense?

**Comment by jimschubert** [comment]
@ejithon definitely! Thanks.

**Pull request merged**